### PR TITLE
Nisha | Stage | Version 2 Varaint wise export report

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvh7cnb

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 1]
```diff
@@ -41,16 +41,124 @@ export class InventoryComponentStore extends ComponentStore<any> {
                 return this.inventoryService.getItemWiseReportExport(req.queryParams, req.stockReportRequest).pipe(
                     tapResponse(
                         (res: BaseResponse<any, any>) => {
-                            if (res.status === "success") {
-                                if (typeof res?.body === "string") {
-                                    this.toaster.showSnackBar("success", res?.body);
+                            if (res.status === "success" && typeof res?.body === "string") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res?.status === "success" && typeof res?.body === "string") {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 2]
```diff
@@ -41,16 +41,124 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+     * @memberof InventoryComponentStore
+     */
+    readonly exportVariant = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                this.patchState({ isLoading: true });
+                return this.inventoryService.getVariantWiseReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res.status === "success" && typeof res?.body === "string") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res?.status === "success" && typeof res?.body === "string") {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 3]
```diff
@@ -41,16 +41,124 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+     * @memberof InventoryComponentStore
+     */
+    readonly exportGroup = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                this.patchState({ isLoading: true });
+                return this.inventoryService.getGroupWiseReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res.status === "success" && typeof res?.body === "string") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res?.status === "success" && typeof res?.body === "string") {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 4]
```diff
@@ -41,16 +41,124 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+     * @memberof InventoryComponentStore
+     */
+    readonly exportTransaction = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                this.patchState({ isLoading: true });
+                return this.inventoryService.getTransactionReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res.status === "success" && typeof res?.body === "string") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res?.status === "success" && typeof res?.body === "string") {
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**